soh/soh/z_play_otr.cpp

```diff
@@ -29,11 +29,13 @@ extern "C" void OTRPlay_SpawnScene(PlayState* play, s32 sceneNum, s32 spawn) {
 
     //osSyncPrintf("\nSCENE SIZE %fK\n", (scene->sceneFile.vromEnd - scene->sceneFile.vromStart) / 1024.0f);
 
-    std::string sceneVersion;
-    if (IsGameMasterQuest()) {
-        sceneVersion = "mq";
-    } else {
-        sceneVersion = "nonmq";
+    // Scenes considered "dungeon" with a MQ variant
+    int16_t inNonSharedScene = (sceneNum >= SCENE_DEKU_TREE && sceneNum <= SCENE_ICE_CAVERN) ||
+                               sceneNum == SCENE_GERUDO_TRAINING_GROUND || sceneNum == SCENE_INSIDE_GANONS_CASTLE;
```

Comment:
i feel like we do something similar in quite a few places throughout the codebase. i know changing the scene order would break things, but i think it'd be nice to come up with a reusable reliable source of truth for stuff like "is this scene a dungeon" and "is this scene the same in mq and vanilla"

that feels like code cleanup and out of scope for this PR though

Reply:
Yeah, some of them are dungeons + bosses, or all of ganons scenes instead of just the trials part. So those aren't useful here. But agree overall that we can try to clean up and move those to a similar location.